### PR TITLE
Lift all dependencies (the big one)

### PR DESCRIPTION
**Pull request by ggwpez**
After preparing in https://github.com/paritytech/polkadot-sdk/pull/4633, we can lift also all internal dependencies up to the workspace.

This does not actually change anything, but uses `workspace = true` for all dependencies. You can check it with:  
```bash
git checkout -q $(git merge-base oty-lift-all-deps origin/master)
cargo tree -e features > master.out

git checkout -q oty-lift-all-deps
cargo tree -e features > new.out
diff master.out new.out
```

It did not yet lift 100% of dependencies, some inside of `target.*` or some that had conflicting aliases introduced recently. But i will do these together in a follow-up with CI checks.

Can be reproduced with [zepter](https://github.com/ggwpez/zepter/): `zepter transpose d lift-to-workspace "regex:.*" --version-resolver highest --skip-package "polkadot-sdk" --ignore-errors --fix`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -577,6 +577,768 @@ serde_yaml = { version = "0.9" }
...
+pallet-sudo = { path = "substrate/frame/sudo", default-features = false }
+pallet-template = { path = "templates/solochain/pallets/template", default-features = false }
+pallet-timestamp = { path = "substrate/frame/timestamp", default-features = false }
+pallet-tips = { path = "substrate/frame/tips", default-features = false }
+pallet-transaction-payment = { path = "substrate/frame/transaction-payment", default-features = false }
+pallet-transaction-payment-rpc = { path = "substrate/frame/transaction-payment/rpc", default-features = false }
+pallet-transaction-payment-rpc-runtime-api = { path = "substrate/frame/transaction-payment/rpc/runtime-api", default-features = false }
+pallet-transaction-storage = { version = "*", default-features = false }
+pallet-treasury = { path = "substrate/frame/treasury", default-features = false }
+pallet-tx-pause = { version = "*", default-features = false }
```
Comment (bkchr) [inline Cargo.toml:967]: Looks wrong :P

File: Cargo.toml [hunk 2]
```diff
@@ -577,6 +577,768 @@ serde_yaml = { version = "0.9" }
...
+pallet-state-trie-migration = { path = "substrate/frame/state-trie-migration", default-features = false }
+pallet-statement = { version = "*", default-features = false }
+pallet-sudo = { path = "substrate/frame/sudo", default-features = false }
+pallet-template = { path = "templates/solochain/pallets/template", default-features = false }
+pallet-timestamp = { path = "substrate/frame/timestamp", default-features = false }
+pallet-tips = { path = "substrate/frame/tips", default-features = false }
+pallet-transaction-payment = { path = "substrate/frame/transaction-payment", default-features = false }
+pallet-transaction-payment-rpc = { path = "substrate/frame/transaction-payment/rpc", default-features = false }
+pallet-transaction-payment-rpc-runtime-api = { path = "substrate/frame/transaction-payment/rpc/runtime-api", default-features = false }
+pallet-transaction-storage = { version = "*", default-features = false }
```
Comment (bkchr) [inline Cargo.toml:965]: Dito

File: Cargo.toml [hunk 3]
```diff
@@ -577,6 +577,768 @@ serde_yaml = { version = "0.9" }
...
+pallet-session = { path = "substrate/frame/session", default-features = false }
+pallet-session-benchmarking = { path = "substrate/frame/session/benchmarking", default-features = false }
+pallet-skip-feeless-payment = { path = "substrate/frame/transaction-payment/skip-feeless-payment", default-features = false }
+pallet-society = { path = "substrate/frame/society", default-features = false }
+pallet-staking = { path = "substrate/frame/staking", default-features = false }
+pallet-staking-reward-curve = { path = "substrate/frame/staking/reward-curve", default-features = false }
+pallet-staking-reward-fn = { path = "substrate/frame/staking/reward-fn", default-features = false }
+pallet-staking-runtime-api = { path = "substrate/frame/staking/runtime-api", default-features = false }
+pallet-state-trie-migration = { path = "substrate/frame/state-trie-migration", default-features = false }
+pallet-statement = { version = "*", default-features = false }
```
Comment (bkchr) [inline Cargo.toml:957]: Here

File: Cargo.toml [hunk 4]
```diff
@@ -577,6 +577,768 @@ serde_yaml = { version = "0.9" }
...
+sp-consensus-babe = { path = "substrate/primitives/consensus/babe", default-features = false }
+sp-consensus-beefy = { path = "substrate/primitives/consensus/beefy", default-features = false }
+sp-consensus-grandpa = { path = "substrate/primitives/consensus/grandpa", default-features = false }
+sp-consensus-pow = { path = "substrate/primitives/consensus/pow", default-features = false }
+sp-consensus-sassafras = { path = "substrate/primitives/consensus/sassafras", default-features = false }
+sp-consensus-slots = { path = "substrate/primitives/consensus/slots", default-features = false }
+sp-core = { path = "substrate/primitives/core", default-features = false }
+sp-core-hashing = { version = "*", default-features = false }
+sp-core-hashing-proc-macro = { version = "*", default-features = false }
+sp-crypto-ec-utils = { version = "*", default-features = false }
```
Comment (bkchr) [inline Cargo.toml:1207]: Here

File: Cargo.toml [hunk 5]
```diff
@@ -577,6 +577,768 @@ serde_yaml = { version = "0.9" }
...
+pallet-offences-benchmarking = { path = "substrate/frame/offences/benchmarking", default-features = false }
+pallet-paged-list = { path = "substrate/frame/paged-list", default-features = false }
+pallet-parachain-template = { path = "templates/parachain/pallets/template", default-features = false }
+pallet-parameters = { path = "substrate/frame/parameters", default-features = false }
+pallet-preimage = { path = "substrate/frame/preimage", default-features = false }
+pallet-proxy = { path = "substrate/frame/proxy", default-features = false }
+pallet-ranked-collective = { path = "substrate/frame/ranked-collective", default-features = false }
+pallet-recovery = { path = "substrate/frame/recovery", default-features = false }
+pallet-referenda = { path = "substrate/frame/referenda", default-features = false }
+pallet-remark = { version = "*", default-features = false }
```
Comment (bkchr) [inline Cargo.toml:941]: Yep

**Review by bkchr — COMMENTED** [review]
Found some problems :D

File: Cargo.toml [hunk 6]
```diff
@@ -577,6 +577,768 @@ serde_yaml = { version = "0.9" }
 syn = { version = "2.0.53" }
 thiserror = { version = "1.0.48" }
 tracing-subscriber = { version = "0.3.18" }
+Inflector = { version = "0.11.4" }
```
Comment (muharem) [inline Cargo.toml:580]: how about grouping them by internal and external (path and version)?

Comment (ggwpez) [reply Cargo.toml:580]: Yea we could think about groupings, but personally i just put it in here and use the taplo to format them.  
If taplo does not enforce the order, then someone will mess it up inevitably.

File: Cargo.toml (hunk 1 above)
Comment (ggwpez) [reply Cargo.toml]: Yep i added some CI check now.

**Review by bkchr — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

File: Cargo.toml [hunk 7]
```diff
@@ -578,6 +578,769 @@ serde_yaml = { version = "0.9" }
...
+prost-build = { version = "0.12.4" }
+pyroscope = { version = "0.5.7" }
+pyroscope_pprofrs = { version = "0.2.7" }
+quickcheck = { version = "1.0.3", default-features = false }
+rand = { version = "0.8.5", default-features = false }
+rand_chacha = { version = "0.3.1", default-features = false }
+rand_core = { version = "0.6.2" }
+rand_distr = { version = "0.4.3" }
+rand_pcg = { version = "0.3.1" }
+rayon = { version = "1.5.1" }
```
[the remark this answers is not in the document]
Comment (ggwpez) [reply Cargo.toml:1073]: >Would have been nice for all of them, but just a nitty nitpick.

👍 

>Generally some CI check to ensure we always add stuff with workspace = true would be good, but can come later.

There are some checks in place that should prevent the most obvious cases, but i did not yet check yet if its actually bullet-proof.

File: master.wasm
Comment (nazar-pc) [inline master.wasm:1]: This commit added 3 `.wasm` files, two of which are in the root of the repository, yet commit message simply says "Taplo".

Why?

Comment (bkchr) [reply master.wasm:1]: Haha nice find :D https://github.com/paritytech/polkadot-sdk/pull/5052

**Pull request merged**